src/pages/iou/IOUModal.js

```diff
@@ -0,0 +1,173 @@
+import React, {Component} from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+import {redirectToLastReport} from '../../libs/actions/App';
+import IOUAmountPage from './steps/IOUAmountPage';
+import IOUParticipantsPage from './steps/IOUParticipantsPage';
+import IOUConfirmPage from './steps/IOUConfirmPage';
+import Header from '../../components/Header';
+import styles from '../../styles/styles';
+import Icon from '../../components/Icon';
+import getPreferredCurrency from '../../libs/actions/IOU';
+import {Close, BackArrow} from '../../components/Icon/Expensicons';
+
+/**
+ * IOU modal for requesting money and splitting bills.
+ */
+const propTypes = {
+
+    /* Onyx Props */
+    // Url currently in view
+    currentURL: PropTypes.string,
+
+    // IOU modal data
+    iouData: PropTypes.shape({
+        isLoadingCurrency: PropTypes.bool.isRequired,
+        isLoadingParticipants: PropTypes.bool.isRequired,
+        createReportInProgress: PropTypes.bool.isRequired,
+    }),
+};
+
+const StepType = {
+    IOUAmount: 'IOUAmount',
+    IOUParticipants: 'IOUParticipants',
+    IOUConfirm: 'IOUConfirm',
+};
+
+const defaultProps = {
+    currentURL: '',
+    iouData: {
+        isLoadingCurrency: true,
+        isLoadingParticipants: true,
+        createReportInProgress: false,
+    },
+};
+
+class IOUModal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            steps: [StepType.IOUAmount, StepType.IOUParticipants, StepType.IOUConfirm],
+            currentStepIndex: 0,
+            hasMultipleParticipants: this.props.currentURL === ROUTES.IOU_BILL_SPLIT,
```

Comment:
OK, I'm seeing this again here, and I mentioned something about it in the Router, but I think I have a better idea for this and I'll comment on it below.